ansible/roles/quickfeed/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+---
+  - name: Ensure group {{ quickfeed_group }} exists
+    ansible.builtin.group:
+      name: "{{ quickfeed_group }}"
+      state: present
+
+  - name: Create {{ quickfeed_user }} runner user
+    ansible.builtin.user:
+      create_home: true
+      name: "{{ quickfeed_user }}"
+      group: "{{ quickfeed_group }}"
+      shell: /bin/bash
+
+  - name: Setup GOPATH for {{ quickfeed_user }}
+    become_user: "{{ quickfeed_user }}"
+    ansible.builtin.copy:
+      content: |
+        export GOPATH={{ quickfeed_home }}/go
+        export PATH=$PATH:$GOPATH/bin
+        export GO111MODULE=on
+      dest: "{{ quickfeed_home }}/.bashrc"
+      mode: 0644
+
+  - name: Define GOPATH facts
+    become_user: "{{ quickfeed_user }}"
+    ansible.builtin.set_fact:
+      GOPATH: "{{ quickfeed_home }}/go"
+    when: GOPATH is not defined
+  
+  - name: Define PATH facts
+    become_user: "{{ quickfeed_user }}"
+    ansible.builtin.set_fact:
+      PATH: "{{ ansible_env.PATH }}:{{ GOPATH }}/bin"
+    when: GOPATH is defined
+
+  - name: Define GO111MODULE facts
+    become_user: "{{ quickfeed_user }}"
+    ansible.builtin.set_fact:
+      GO111MODULE: "on"
+    when: GO111MODULE is not defined
+
+  # go dependencies
+  - name: Get Go dependencies packages
+    become_user: "{{ quickfeed_user }}"
+    ansible.builtin.command: go get -u {{ item }}
```

Comment:
The recommended way to install go dependencies is to use `go install {{ item }}`.

Reply:
But `go install` will download the package? I thought that the install command just install local packages (i.e. packages that already exists in GOPATH), and that `go get` downloads and installs the packages. And in this case, I need to download the packages.

https://golang.org/pkg/cmd/go/internal/get/#pkg-variables

Reply:
Yes. `go install` will also download the package or command/tool. For example, in this case, I believe it is being used to download and install various tools for grpc and protobuf. This [thread has a nice solution](https://github.com/go-modules-by-example/index/issues/77) for using the `tools.go` approach to specify the dependencies; meaning that make targets like this:

```sh
cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go install %
```
can be replaced with something like this:
```sh
go install $(go list -f "{{range .Imports}}{{.}} {{end}}" tools.go)
```
Running this:
```sh
go list -f "{{range .Imports}}{{.}} {{end}}" tools.go
```
should give you the list of tools to install.

Reply:
Oh! Cool, I didn't know that. I'll do in this way then :)